### ISSUE TITLE
LICH contents reassembly

### ISSUE THREAD

**Issue by sp5wwp** [M17-Project/M17_spec#131]
Link Setup Data* must remain the same over the course of a superframe, and receivers should only reassemble the LSD chunks from within a single superframe.

*naming convention - LSF is the first frame containing LSD. LSD is then also the data reassembled using LICH. Logically, it's not LSF.

**Comment by tarxvftech** [comment]
I see why the desire for having a different name due to the changing data and multiple uses. 
Makes sense, that distinction needs to be made.

So we have:
* LICH which is the data itself
* LSD which ... is the same data? Or is it a single chunk of the LICH?
* LSF which is the actual frame that contains full LICH/LSD sent at the beginning of a new transmission

**Comment by sp5wwp** [comment]
Starting with LSD:
LSD - link setup data that contains the destination, source, metadata. It is only data, represented with bytes.
LSF - it is the first frame containing full LSD. It is frame, a basic logical element of the RF transmission.
LICH - it is the channel within M17 RF transmission, containing chunks of LSD. It is present in every stream frame containing payload. It can be used to reconstruct LSD, if all 6 frames forming a superframe are decoded and the resulting data is concatenated together. LSD can change in time, on superframe basis.

**Comment by tarxvftech** [comment]
Gotcha. If I have that right then each of these should be correct:

LICH refers specifically to the chunks of LSD.
LSF contains the full LSD. 
The LICH stream allows you to assemble the full LSD if you collect a full superframe's worth of chunks.

So we're being more precise with the language. I like it, makes sense.

**Comment by sp5wwp** [comment]
Exactly. It makes little sense to say "recover full LSF" as LSF is a frame, not its contents. Or "LSF has to remain the same over the course of a single superframe" - again, LSF is a frame and not the data it holds.

**Comment by sp5wwp** [comment]
@n7tae Fixed with [808327e](https://github.com/M17-Project/M17_spec/commit/808327eeb861b4df613edc89e0de04b21e95f1df)?

**Comment by n7tae** [comment]
Yes.
